app/controllers/concerns/authentication.rb

```diff
@@ -12,18 +12,22 @@ module Authentication
   ].freeze
 
   included do
-    before_action :authenticate_user
+    around_action :authenticate_user
```

Comment:
I think it would simplify our code and make it easier to understand if we split this into separate `before_action` and `after_action`. i.e. `clear_user`

Reply:
I've experimented with that. The difference is that `after_action` isn't triggered if the action isn't run (e.g. on auth failure or similar).

Reply:
Isn't that OK? If auth fails, the current user would not be set.

Reply:
Here I want to be perfectly sure that it would be cleared out. There might be other callbacks in the registry that can stop the action execution.